src/ports/transak/component.ts

```diff
@@ -0,0 +1,59 @@
+import { AppComponents } from '../../types'
+import { ITransakComponent, OrderResponse } from './types'
+
+export function createTransakComponent(
+  components: Pick<AppComponents, 'fetch'>,
+  {
+    apiURL,
+    apiKey,
+    apiSecret
+  }: {
+    apiURL: string
+    apiKey: string
+    apiSecret: string
+  }
+): ITransakComponent {
+  const { fetch } = components
+  let accessToken: string
+  let accessTokenExpiresAt: number
+  /**
+   * Given the order id, returns relevant data related to status changes (status & tx hash).
+   *
+   * @param orderId - Transak Order ID.
+   */
+  async function getOrder(orderId: string): Promise<OrderResponse> {
+    if (!accessToken || Date.now() > Number(`${accessTokenExpiresAt}000`)) {
```

Comment:
Mmm I think this is prone to errors. 

If you call the `getOrder` method:

1. The first part of the condition is false: `!accessToken` because `accessToken` is `undefined`.
2. Then the second part is evaluated: `Date.now() > Number(`${accessTokenExpiresAt}000`)`, here `accessTokenExpiresAt` is also `undefined`, so the string evaluated is `undefined000`. Then that results in `NaN` after the `Number()` wrapper. Finally the condition is false, because `Date.now() > NaN` is always false.

I think it would be better to initialize the values of the variables above so they are not `undefined`, and then you can treat `accessTokenExpiresAt`  always like a number without the double casting to string and back to number

```suggestion
    if (!accessToken || !accessTokenExpiresAt || Date.now() > accessTokenExpiresAt * 1000) {
```